packages/type-extrinsics/src/SignableExtrinsic.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright 2017-2018 @polkadot/extrinsics authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { Extrinsic, Index, UncheckedExtrinsic } from '@polkadot/api-codec/index';
+import { KeyringPair } from '@polkadot/util-keyring/types';
+import u8aConcat from '@polkadot/util/u8a/concat';
+
+export default class SignableExtrinsic extends UncheckedExtrinsic {
+  sign (sender: KeyringPair, nonce: Index): Extrinsic {
+    // Prepend sender's publicKey and nonce
+    const message = u8aConcat(
+      sender.publicKey(),
+      nonce.toU8a(),
+      this.toU8a(true)
+    );
+
+    const signature = sender.sign(message);
+
+    return new Extrinsic(
+      u8aConcat(
+        new Uint8Array([255]), // FIXME works, but why?
```

Comment:
ideally `sender.address().prefixKey()`

Reply:
It is the prefix for AccountId. So if you had it as `Address` and then used the `toU8a()` it would have added the prefix. Maybe there is a much-up with AccountId as well - I know that AccountIndex is not correct, it itself should add the prefix.

Reply:
TL;DR Making it an Address and encoding it will add the prefix (Should be Address, so it handles both AccountId & AccountIndex)